source/common/upstream/cluster_manager_impl.cc

```diff
@@ -592,7 +592,8 @@ void ClusterManagerImpl::loadCluster(const envoy::api::v2::Cluster& cluster,
   // finishes.
   if (cluster_reference.info()->lbType() == LoadBalancerType::RingHash) {
     cluster_entry_it->second->thread_aware_lb_ = std::make_unique<RingHashLoadBalancer>(
-        cluster_reference.prioritySet(), cluster_reference.info()->stats(), runtime_, random_,
+        cluster_reference.prioritySet(), cluster_reference.info()->stats(),
```

Comment:
Arguably you could also just add the ring hash LB stats to `ClusterStats`; we have both zone aware and subset LB stats crammed in there today. I think it's a bit cleaner to plumb the stats scope though.

Reply:
Yeah, I figure it's worth the extra bit of plumbing to have the cleaner nested scope, so if the user isn't using ring hash lb, they don't see these extra gauges and wonder why they're pinned at 0.